lib/ecto/type.ex

```diff
@@ -76,6 +76,14 @@ defmodule Ecto.Type do
 
   import Kernel, except: [match?: 2]
 
+  alias Ecto.Types.DateType
```

Comment:
Let's please keep the logic for this inlined in this single module instead of introducing a bunch of new modules.